server/src/QRServer/common/classes.py

```diff
@@ -112,17 +118,27 @@ def generate_match_report(self) -> Optional[DbMatchReport]:
         p2_id = list(self.match_stats.keys())[1]
         stat1 = self.match_stats[p1_id]
         stat2 = self.match_stats[p2_id]
-        winner_id, winner, loser_id, loser = (p1_id, stat1, p2_id, stat2) if \
-            stat1.own_piece_count > stat2.own_piece_count else (p2_id, stat2, p1_id, stat2)
+
+        if not stat1:
```

Comment:
this looks quite ugly, we could get rid of it if we only used winner's report. If we decide that we want to do so, then we could most likely ignore almost every other change in this PR, or simplify it greatly, since we only need winners' stats